### PR TITLE
Issue AWS Infrastructure

### PR DESCRIPTION
**Pull request by thefishua**
Structure for AWS issues based off architecture diagram with a SNS  -> SQS -> Lambda interface with the SNS connected to the s3 bucket openapi_files_bucket

### CODE REVIEW

File: terraform/aws_only/issues.tf [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+      Resource = "${aws_sqs_queue.github_issues_queue.arn}"
+      Principal = {
+        Service = "lambda.amazonaws.com"
+      }
+    }]
+  })
+}
+
+# # Github Issues Lambda
+# Lambda function policy
```
Comment (The-TT-Hacker) [inline terraform/aws_only/issues.tf]: you dont need these comments, the name of the resource should be enough

File: terraform/aws_only/issues.tf [hunk 2]
```diff
@@ -0,0 +1,262 @@
+# Github Issues Architecture
```
Comment (The-TT-Hacker) [inline terraform/aws_only/issues.tf]: this stuff will work in localstack, it doesn't need to be in aws_only

File: internal/platform/platform_test.go [hunk 3]
```diff
@@ -46,40 +46,40 @@ func TestFormatFuzzBody(t *testing.T) {
...
-		"- Accept: application/json\n" +
-		"- Host: localhost:8888\n" +
-		"- Content-Type: application/json\n" +
-		"- Request: {    \"id\":99,    \"body\":\"my first blog post\"}\n\n" +
-		"! producer_timing_delay 0\n" +
-		"! max_async_wait_time 20\n" +
-		"\nPREVIOUS RESPONSE: 'HTTP/1.1 201 Created response:{\"id\":10,\"body\":\"my first blog post\"}'\n"
-
+	expectedBody := "# InvalidDynamicObjectChecker Invalid 20x Response\n" + 
+									"\nDetails: Detects 500 errors or unexpected success status codes when invalid dynamic objects are sent in requests.\n" +
```
Comment (The-TT-Hacker) [inline internal/platform/platform_test.go]: why these huge spaces?

File: internal/platform/platform_test.go [hunk 4]
```diff
@@ -22,20 +22,20 @@ func TestFormatFuzzBody(t *testing.T) {
 	errorType := "InvalidDynamicObjectChecker"
 	methodInformation := result.DynoMethodInformation{
 		AcceptedResponse: &acceptedResponse,
-		Host:             &host,
-		ContentType:      &contentType,
-		Request:          &request,
+		Host: &host,
```
Comment (The-TT-Hacker) [inline internal/platform/platform_test.go]: can you remove all the white spaces after each of these lines. its annoying gofmt doesn't do this stuff. I want to figure out how to make it stricter

File: terraform/aws_only/issues.tf (hunk 2 above)
Comment (thefishua) [reply terraform/aws_only/issues.tf]: think i got confused with the naming of the folders thought it was the reverse

File: internal/platform/platform_test.go (hunk 4 above)
Comment (thefishua) [reply internal/platform/platform_test.go]: will do

File: terraform/aws_only/issues.tf [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+          "Condition":{
+              "ArnLike":{"aws:SourceArn":"${aws_s3_bucket.openapi_files_bucket.arn}"}
+          }
+        }
+      ]
+    }
+  POLICY
+}
+# Event for everytime an Object/Fuzz Results is created will notify SNS
+resource "aws_s3_bucket_notification" "s3_notif" {
```
Comment (The-TT-Hacker) [inline terraform/aws_only/issues.tf]: this resource needs a better name

**Comment by thefishua** [comment]
Don't merge the branch yet as waiting on @stevensnicholas to fix main

File: terraform/main/issues.tf [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+            "aws:SourceArn": "${aws_sns_topic.results_updates.arn}"
+          }
+        }
+      }
+    ]
+  }
+  POLICY
+}
+
+# SQS Policy for Lambda Function
```
Comment (The-TT-Hacker) [inline terraform/main/issues.tf]: if the comment doesn't add anything that cant be put in the name of the resource, remove it. do this for all resources in this PR

File: terraform/main/issues.tf [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+          "ArnLike": {
+            "aws:SourceArn": "${aws_s3_bucket.openapi_files_bucket.id}"
+          }
+        }
+      }
+    ]
+  }
+  POLICY
+  tags = {
+    name = "github_issues_queue"
```
Comment (The-TT-Hacker) [inline terraform/main/issues.tf]: dont think you need this tag

File: terraform/main/issues.tf (hunk 6 above)
Comment (thefishua) [reply terraform/main/issues.tf]: On it

**Comment by The-TT-Hacker** [comment]
have you tried running the terraform locally?

**Comment by thefishua** [comment]
im doing that now with `terraform apply`, but im running into issues else where

File: terraform/main/issues.tf [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+    actions = [
+      "kms:GenerateDataKey",
+      "kms:Decrypt"
+    ]
+    resources = ["${aws_s3_bucket.bucket.arn}"]
+  }
+  statement {
+    effect = "Allow"
+    principals {
+      identifiers = ["arn:aws:iam::${account_id}:root"]
```
Comment (The-TT-Hacker) [inline terraform/main/issues.tf:63]: what is account_id?

**Pull request closed without merging**